File: Data Structures and Algorithms/Python/SparseMatrix.py
#
# Ayush Kumar (202103103510253)
# 4th Sem Div: A
# SparseMatrix.py
#

class SparseMatrix:
  __matrix = [[], [], []]
  def __init__(self, matrix):
    for row in range(len(matrix)):
      for column in range(len(matrix[0])):
        if(matrix[row][column] != 0):
          self.__matrix[0].append(row)
          self.__matrix[1].append(column)
          self.__matrix[2].append(matrix[row][column])
          
    for row in self.__matrix:
      print("|", end =" ")
      for element in row:
        print(element, end = " ")
      print("|")

File: Data Structures and Algorithms/Java/SparseMatrix.java
/*
    Ayush Kumar (202103103510253)
    4th Sem Div: A
    SparseMatrix.java
*/

package include;

public class SparseMatrix {
    int[][] matrix;
    
    public SparseMatrix(int[][] matrix) {
      int size = 0;
      for(int row = 0; row < matrix.length; row++) {
        for(int column = 0; column < matrix[0].length; column++) {
          if(matrix[row][column] != 0) {
            size++;
          }
        }
      }
  
      this.matrix = new int[3][size];
  
      int index = 0;
      for(int row = 0; row < matrix.length; row++) {
        for(int column = 0; column < matrix[0].length; column++) {
          if(matrix[row][column] != 0) {
            this.matrix[0][index] = row;
            this.matrix[1][index] = column;
            this.matrix[2][index] = matrix[row][column];
            index++;
          }
        }
      }
  
      for(int row = 0; row < this.matrix.length; row++) {
        System.out.print("| ");
        for(int column = 0; column < this.matrix[0].length; column++) {
          System.out.print(this.matrix[row][column] + " ");
        }
        System.out.println("|");
      }
    }
  }
  